src/api/app/assets/javascripts/webui2/kiwi_editor.js

```diff
@@ -0,0 +1,400 @@
+  canSave = true;
+  $('#kiwi-image-update-form-save, #kiwi-image-update-form-revert').removeClass('disabled');
+}
+
+function editPackageDialog(){
+  var fields = $(this).parents('.nested-fields');
+  var dialog = fields.find('.modal');
+  dialog.modal('show');
+}
+
+function editRepositoryDialog(){
+  var fields = $(this).parents('.nested-fields');
+  var dialog = fields.find('.modal');
+  var sourcePath = fields.find("[id$='source_path']");
+
+  dialog.modal('show');
+  var matchedObsSourcePath = sourcePath.val().match(/^obs:\/\/([^\/]+)\/([^\/]+)$/);
+  if (matchedObsSourcePath) {
+    var projectField = fields.find('[name=target_project]');
+    var repoField = fields.find('[name=target_repo]');
+    var aliasField = fields.find('[name=alias_for_repo]');
+    var expertRepoAlias = fields.find("[id$='alias']");
+    var repoTypeField = fields.find("[id$='repo_type']");
+    aliasField.val(expertRepoAlias.val());
+    projectField.val(matchedObsSourcePath[1]);
+    repoField.html('');
+    repoField.append(new Option(matchedObsSourcePath[2]));
+    repoField.val(matchedObsSourcePath[2]);
+    repoTypeField.val('rpm-md');
+  }
+
+  $('span[role=status]').text(''); // empty autocomplete status
+}
+
+function addRepositoryErrorMessage(sourcePath, field) {
+  if (sourcePath === 'obsrepositories:/') {
+    field.text('If you want use obsrepositories:/ as source_path, please check the checkbox "use project repositories".');
+  }
+  else {
+    field.text('The source path can not be empty!');
+  }
+
+  field.removeClass('d-none');
+}
+
+function closeDescriptionDialog() {
+  var fields = $(this).parents('.nested-fields');
+  var dialog = fields.find('.modal');
+  var name = dialog.find("[id$='name']");
+
+  if (name.val() !== '') {
+    $('#image-name').text(name.val());
+  }
+  else {
+    fields.find(".ui-state-error").removeClass('d-none');
+    return false;
+  }
+
+  var elements = fields.find('.fill');
+  for(var i=0; i < elements.length; i++) {
+    var object = dialog.find("[id$='" + $(elements[i]).data('tag') + "']");
+    if ( object.val() !== "") {
+      $(elements[i]).text(object.val());
+    }
+  }
+
+  addDefault(dialog);
+
+  if (!canSave) {
+    enableSave();
+  }
+
+  fields.find(".ui-state-error").addClass('d-none');
+
+  hideOverlay(dialog);
+}
+
+function closePreferencesDialog() {
+  var fields = $(this).parents('.nested-fields');
+  var dialog = fields.find('.modal');
+
+  var elements = fields.find('.fill');
+  for(var i=0; i < elements.length; i++) {
+    var object = dialog.find("[id$='_" + i + "_" + $(elements[i]).data('tag') + "']");
+    if ( object.val() !== "") {
+      if ( $(elements[i]).data('tag') === 'type_image' ) {
+        $(elements[i]).text(object.find(":selected").text());
+      }
+      else {
+        $(elements[i]).text(object.val());
+      }
+    }
+  }
+
+  addDefault(dialog);
+
+  if (!canSave) {
+    enableSave();
+  }
+
+  hideOverlay(dialog);
+}
+
+
+
+function closeDialog() {
```

Comment:
remember that all javascript functions are in one namespace for all routes. So function names like closeDialog will clash super easily.

Reply:
this comment from @coolo is important. Could you rename to closeKiwiDialog or something like that? :smile: